test/registration/test_ndt.cpp

```diff
@@ -46,52 +46,60 @@
 
 using namespace pcl;
 using namespace pcl::io;
- 
-PointCloud<PointXYZ> cloud_source, cloud_target;
- 
-//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-TEST (PCL, NormalDistributionsTransform)
-{
-  using PointT = PointNormal;
-  PointCloud<PointT>::Ptr src (new PointCloud<PointT>);
-  copyPointCloud (cloud_source, *src);
-  PointCloud<PointT>::Ptr tgt (new PointCloud<PointT>);
-  copyPointCloud (cloud_target, *tgt);
-  PointCloud<PointT> output;
 
-  NormalDistributionsTransform<PointT, PointT> reg;
+PointCloud<PointXYZ>::Ptr cloud_source, cloud_target;
+
+void test_ndt(NormalDistributionsTransform<PointXYZ, PointXYZ>& reg) {
   reg.setStepSize (0.05);
   reg.setResolution (0.025f);
-  reg.setInputSource (src);
-  reg.setInputTarget (tgt);
+  reg.setInputSource (cloud_source);
+  reg.setInputTarget (cloud_target);
   reg.setMaximumIterations (50);
   reg.setTransformationEpsilon (1e-8);
+
   // Register
+  PointCloud<PointXYZ> output;
   reg.align (output);
-  EXPECT_EQ (output.size (), cloud_source.size ());
+  EXPECT_EQ (output.size (), cloud_source->size ());
   EXPECT_LT (reg.getFitnessScore (), 0.001);
+  Eigen::Matrix4f transform = reg.getFinalTransformation();
 
-  // Check again, for all possible caching schemes
-  for (int iter = 0; iter < 4; iter++)
-  {
-    bool force_cache = (bool) iter/2;
-    bool force_cache_reciprocal = (bool) iter%2;
-    pcl::search::KdTree<PointT>::Ptr tree(new pcl::search::KdTree<PointT>);
-    // Ensure that, when force_cache is not set, we are robust to the wrong input
-    if (force_cache)
-      tree->setInputCloud (tgt);
-    reg.setSearchMethodTarget (tree, force_cache);
+  // Check if the single thread result is consistent
+  reg.setNumberOfThreads(4);
+  reg.align(output);
+  EXPECT_LT((reg.getFinalTransformation() - transform).array().abs().maxCoeff(), 1e-6);
+}
 
-    pcl::search::KdTree<PointT>::Ptr tree_recip (new pcl::search::KdTree<PointT>);
-    if (force_cache_reciprocal)
-      tree_recip->setInputCloud (src);
-    reg.setSearchMethodSource (tree_recip, force_cache_reciprocal);
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST (PCL, NormalDistributionsTransform_KDTREE)
```

Comment:
This looks like a job for https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests

Reply:
Done